File: tests/fixtures/sample.ts
interface Request {
    url: string;
    method: string;
}

function makeWebRequest(req: WebRequest): void {
    req.send();
}

function makeHttpRequest(req: HttpRequest): void {
    req.execute();
}

class WebRequest implements Request {
    url: string;
    method: string = "GET";

    constructor(url: string) {
        this.url = url;
    }

    send(): Promise<Response> {
        return fetch(this.url);
    }
}

class HttpRequest implements Request {
    url: string;
    method: string = "POST";

    constructor(url: string) {
        this.url = url;
    }

    execute(): Promise<Response> {
        return fetch(this.url, { method: this.method });
    }
}

function acceptsString(input: string): string {
    return input.trim();
}

File: tests/fixtures/sample.py
import os
from typing import Optional


def greet(name: str) -> str:
    return f"Hello, {name}!"


def add(a: int, b: int) -> int:
    return a + b


class HttpClient:
    def fetch(self, url: str) -> bytes:
        pass

    def post(self, url: str, data: str) -> bytes:
        pass


class WebRequest:
    def send(self, endpoint: str) -> None:
        pass


def use_web_request(req: WebRequest) -> None:
    req.send("/api")


def use_http_client(client: HttpClient) -> None:
    client.fetch("https://example.com")
